### PR TITLE
Move PageCacheRecycler into BigArrays

### PR DESCRIPTION
**Pull request by s1monw**
PageCacheRecycler is really just an implementation detail of
BigArrays. There is no need to leak this class anywhere outside of it.

Relates to #18664

### CODE REVIEW

**Comment by s1monw** [comment]
@jasontedor can you take a look

File: core/src/main/java/org/elasticsearch/common/util/PageCacheRecycler.java [hunk 1]
```diff
@@ -63,14 +64,10 @@
...
     @Override
     public void close() {
-        bytePage.close();
-        intPage.close();
-        longPage.close();
-        objectPage.close();
+        Releasables.close(true, bytePage, intPage, longPage, objectPage);
     }
 
-    @Inject
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/util/PageCacheRecycler.java]: Another one bites the dust.

File: test/framework/src/main/java/org/elasticsearch/cache/recycler/MockPageCacheRecycler.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.recycler.Recycler.V;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.PageCacheRecycler;
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/cache/recycler/MockPageCacheRecycler.java]: Maybe move `MockPageCacheRecycler` to `org.elasticsearch.common.util` too?

**Comment by jasontedor** [comment]
@s1monw I left a minor suggestion, otherwise LGTM. Fire at will.

**Pull request merged**